lib/cpp/Kconfig

```diff
@@ -54,6 +56,55 @@ config STD_CPP2B
 
 endchoice
 
+config REQUIRES_FULL_LIBCPP
+	bool
+	select REQUIRES_FULL_LIBC
+	help
+	  Helper symbol to indicate that a full C++ standard library
+	  implementation is required.
+
+	  Select this from a symbol that enables a feature requiring a full
+	  C++ standard library implementation.
+
+choice LIBCPP_IMPLEMENTATION
+	prompt "C++ Standard Library Implementation"
+	default EXTERNAL_LIBCPP if REQUIRES_FULL_LIBCPP && NATIVE_APPLICATION
+	default GLIBCXX_LIBCPP if REQUIRES_FULL_LIBCPP
+	default MINIMAL_LIBCPP
+
+config MINIMAL_LIBCPP
+	bool "Minimal C++ Library"
+	depends on !REQUIRES_FULL_LIBCPP
+	help
+	  Build with the minimal C++ library provided by Zephyr.
+
+	  The Zephyr minimal C++ library only provides a very limited subset
+	  of the standard C++ library and is mainly intended for use with the
+	  applications that do not require the Standard Template Library (STL).
+
+config GLIBCXX_LIBCPP
+	bool "GNU C++ Standard Library"
+	depends on !NATIVE_APPLICATION
+	depends on NEWLIB_LIBC
```

Comment:
Not supported by `picolibc`?

Reply:
See https://github.com/zephyrproject-rtos/zephyr/pull/52946#issuecomment-1347896933